plugins/leemons-plugin-leebrary/frontend/src/components/AssetList.js

```diff
@@ -71,283 +52,227 @@ function getOwner(asset) {
-    stateFilter: null,
-  };
-  const lastQuery = React.useRef(null);
-  const [store, setStoreValue] = useAssetListStore(initialState);
-
-  const [t, translations] = useTranslateLoader(prefixPN('list'));
-  const [, , , getErrorMessage] = useRequestErrorMessage();
-  const [containerRef, containerRect] = useResizeObserver();
   const [childRef, childRect] = useResizeObserver();
+
   const {
     openConfirmationModal,
     openDeleteConfirmationModal,
     setLoading: setAppLoading,
   } = useLayout();
-  const [searchDebounced] = useDebouncedValue(store.searchCriteria, 300);
-  const session = useSession();
-  const locale = getLocale(session);
-  const showThumbnails = useMemo(
-    () => onlyThumbnails && store.category?.key === 'media-files',
-    [onlyThumbnails, store.category]
-  );
 
-  const isEmbedded = useMemo(() => variant === 'embedded', [variant]);
+  // -------------------------------------------------------------------------------------
+  // QUERY HOOKS
+
+  const queryClient = useQueryClient();
+  const assetListQuery = useMemo(() => {
+    setPage(1);
+    const waitForAcademicFilters = allowAcademicFilter && !academicFilters;
+    if (!category || waitForAcademicFilters) return null;
+
+    const query = {
+      providerQuery: academicFilters ? JSON.stringify(academicFilters) : null,
+      criteria: searchCriteria || '',
+      pinned: category?.key === 'pins',
+      category: category?.id,
+      published: allowStatusFilter ? statusFilter : true,
+      showPublic: category?.key === 'pins',
+      preferCurrent: true,
+    };
+
+    if (allowMediaTypeFilter && mediaTypeFilter !== 'all') query.type = mediaTypeFilter;
+
+    if (category?.key.startsWith('leebrary-subject')) {
+      delete query.category;
+      const subjects = isArray(category.id) ? category.id : [category.id];
+      query.subjects = JSON.stringify(subjects);
+    }
+
+    // HANDLE MULTI-CATEGORY SECTIONS
+    if (category?.key === 'leebrary-shared') {
+      delete query.category;
+      query.onlyShared = true;
+    }
+    if (category?.key === 'leebrary-recent') {
+      query.roles = JSON.stringify(['owner']);
+      delete query.category;
+      if (categoryFilter !== 'all') {
+        query.categoryFilter = find(categories, { key: categoryFilter })?.id;
+      }
+    }
+    if (category?.key === 'pins') {
+      delete query.category;
+      if (categoryFilter !== 'all') {
+        query.categoryFilter = find(categories, { key: categoryFilter })?.id;
+      }
+    }
+
+    return query;
+  }, [category, searchCriteria, statusFilter, categoryFilter, mediaTypeFilter, academicFilters]);
+
+  const { data: assetList, isLoading: assetListIsLoading } = useSimpleAssetList({
+    query: assetListQuery,
+    enabled: !!assetListQuery,
+  });
 
   const { items: currentPageAssets } = getPageItems({
-    data: store.assets,
-    page: store.page - 1,
-    size: store.size,
+    data: assetList,
+    page: page - 1,
+    size: pageSize,
   });
```

Comment:
It should be inside a useMemo, to avoid re-renders due to new refence on items (items.slice inside).